### PR TITLE
Feature/orphan node grouping

### PR DESCRIPTION
**Pull request by jayqi**
WIP for #28

### CODE REVIEW

**Comment by jayqi** [comment]
Remaining issues:
- Orphan Node grouping still looks terrible with the layouts
- `tree' layout has a mysterious bug where regenerating the plot will alternate between no-layout and tree-layout. For some reason this doesn't happen with `circle`. 

Code to test
```
myFuncReporter <- PackageFunctionReporter$new()
myFuncReporter$set_package('lubridate')
myFuncReporter$orphanNodeClusteringThreshold <- 3
myFuncReporter$graphLayoutType <- 'tree'
myFuncReporter$plot_network()
```

**Comment by jayqi** [comment]
lubridate_tree
lubridate_no-layout
lubridate_circle

**Comment by bburns632** [comment]
Hey @jayqi ,

I see the function plots looking terrible as a separate issue at the moment than orphan grouping.  I would support merging this PR today (after updates to pass tests) and fixing the graph layouts in a later PR soon after.

**Comment by jayqi** [comment]
Weirdly it looks weird with `lubridate` but it works fine with something smaller like `uptasticsearch`. /shrug
```
myFuncReporter <- PackageFunctionReporter$new()
myFuncReporter$set_package('uptasticsearch')
myFuncReporter$orphanNodeClusteringThreshold <- 3
myFuncReporter$graphLayoutType <- 'tree'
myFuncReporter$plot_network()
```
screen shot 2018-02-09 at 6 43 22 pm

File: tests/testthat/test-AbstractPackageReporter.R [hunk 1]
```diff
@@ -51,7 +51,9 @@ test_that('AbstractPackageReporter structure is as expected', {
   expect_named(object = AbstractPackageReporter$private_fields
                , expected = c(
                  "packageName", 
-                 "packagePath"
+                 "packagePath",
+                 "cache",
+                 "defaultCache"
```
Comment (bburns632) [inline tests/testthat/test-AbstractPackageReporter.R]: If only real life cash multiplied like `pkgnet` cache ...

File: R/AbstractGraphReporter.R [hunk 2]
```diff
@@ -449,6 +483,57 @@ AbstractGraphReporter <- R6::R6Class(
...
+        # Identify orphan nodes
+        identify_orphan_nodes = function() {
+            orphanNodes <- base::setdiff(self$nodes[, node]
+                                         , unique(c(self$edges[, SOURCE], self$edges[, TARGET]))
+                                        )
+            # If there are none, then will be character(0)
+            return(orphanNodes)
+        },
+        
+        graph_layout_functions = list(
```
Comment (bburns632) [inline R/AbstractGraphReporter.R]: I like this.   Clean and structured.

**Comment by bburns632** [comment]
@jayqi Looks good.  I'm going to merge it.  

For posterity, the example to run for this is: 
```
myFuncReporter <- PackageFunctionReporter$new()
myFuncReporter$set_package('uptasticsearch')
myFuncReporter$orphanNodeClusteringThreshold <- 3
#myFuncReporter$graphLayoutType <- 'tree'
myFuncReporter$layoutType <- 'tree'
myFuncReporter$plot_network()
```

**Pull request merged**